server/src/main/java/org/elasticsearch/action/DocWriteRequest.java

```diff
@@ -256,4 +257,20 @@ static ActionRequestValidationException validateSeqNoBasedCASParams(
 
         return validationException;
     }
+
+    static long writeSizeInBytes(Stream<DocWriteRequest<?>> requestStream) {
```

Comment:
Just to clarify what I meant here (@henningandersen asked), I was refering to utility methods such as `RamUsageEstimator.shallowSizeOfInstance` and others here that can give a more accurate estimate of how much memory an object is actually taking up.

Reply:
Is this method still useful now that we use the accountable interface? Let's have BulkShardRequest implement Accountable as well, and then use that directly instead of this method